api/chat_main_handler.go

```diff
@@ -310,47 +287,7 @@ func (h *ChatHandler) OpenAIChatCompletionAPIWithStreamHandler(w http.ResponseWr
 
 }
 
-func getAskMessages(h *ChatHandler, ctx context.Context, w http.ResponseWriter, chatSession sqlc_queries.ChatSession, chatUuid string, regenerate bool) ([]openai.ChatCompletionMessage, error) {
-	chatSessionUuid := chatSession.Uuid
-
-	lastN := chatSession.MaxLength
-	if chatSession.MaxLength == 0 {
-		lastN = 10
-	}
-
-	chat_prompts, err := h.chatService.q.GetChatPromptsBySessionUUID(ctx, chatSessionUuid)
-
-	if err != nil {
-		return nil, fmt.Errorf("fail to get prompt: %w", err)
-	}
-
-	var chat_massages []sqlc_queries.ChatMessage
-	if regenerate {
-		chat_massages, err = h.chatService.q.GetLastNChatMessages(ctx,
-			sqlc_queries.GetLastNChatMessagesParams{
-				Uuid:  chatUuid,
-				Limit: lastN,
-			})
-
-	} else {
-		chat_massages, err = h.chatService.q.GetLatestMessagesBySessionUUID(ctx,
-			sqlc_queries.GetLatestMessagesBySessionUUIDParams{ChatSessionUuid: chatSession.Uuid, Limit: lastN})
-	}
-
-	if err != nil {
-		return nil, fmt.Errorf("fail to get messages: %w", err)
-	}
-	chat_prompt_msgs := lo.Map(chat_prompts, func(m sqlc_queries.ChatPrompt, _ int) openai.ChatCompletionMessage {
-		return openai.ChatCompletionMessage{Role: m.Role, Content: m.Content}
-	})
-	chat_message_msgs := lo.Map(chat_massages, func(m sqlc_queries.ChatMessage, _ int) openai.ChatCompletionMessage {
-		return openai.ChatCompletionMessage{Role: m.Role, Content: m.Content}
-	})
-	msgs := append(chat_prompt_msgs, chat_message_msgs...)
-	return msgs, nil
-}
-
-func chat_stream(ctx context.Context, chatSession sqlc_queries.ChatSession, chat_compeletion_messages []openai.ChatCompletionMessage, w http.ResponseWriter) (string, string, bool) {
+func chat_stream(w http.ResponseWriter, chatSession sqlc_queries.ChatSession, chat_compeletion_messages []openai.ChatCompletionMessage) (string, string, bool) {
 	apiKey := appConfig.OPENAI.API_KEY
 
 	client := openai.NewClient(apiKey)
```

Comment:
:robot: OpenAI

This patch removes the `getAskMessages` function and updates the call to `chat_stream` to remove the `ctx` argument.

Instead of using `getAskMessages`, the `OpenAIChatCompletionAPIWithStreamHandler` function now calls the `GetChatPromptsBySessionUUID` and `GetLatestMessagesBySessionUUID` functions directly on the `chatService` instance.

This change is an improvement because it simplifies the code and makes it more modular.

Please confirm that the `chatService` instance is initialized correctly and implements the necessary functions before merging this patch.

Also, please update the comments on the pull request to reflect this change.

<!-- This is an auto-generated comment by OpenAI -->